JavaScript/chapter02/p04_partition/avc278.js

```diff
@@ -0,0 +1,81 @@
+// Partition: Write code to partition a linked list around a value x, such that all nodes less than x come before all
+// nodes greater than or equal to x. If x is contained within the list, the values of x only need to be after the
+// elements less than x (see below). The partition element x can appear anywhere in the "right partition"; it does not
+// need to appear between the left and right partitions.
+// EXAMPLE
+// Input: 3 -> 5 -> 8 -> 5 -> 10 -> 2 -> 1[partition=5]
+// Output: 3 -> 1 -> 2 -> 10 -> 5 -> 5 -> 8
+
+const assert = require("assert");
+const {
+  arrayToLinkedList,
+  compareLinkedLists,
+} = require("../../lib/avc278/linkedlist");
+
+/**
+ * Partitions a linked list based on the input partition, with lesser values to the left and greater values to the right
```

Comment:
JavaScript does not have a way to express whether a function will mutate the arguments or not (i.e. if it is considered to be [destructive](https://www.cs.cmu.edu/~112/notes/notes-1d-lists.html#destructiveFunctions)), so that needs to be documented.

non-destructive functions typically have slightly different requirements, since they need to make a copy of the input list (which translates to them typically requiring more memory and being a bit slower). therefore, they are not necessarily better (or worse) than their destructive counterparts.